src/tools/yarprobotstatepublisher/src/main.cpp

```diff
@@ -35,9 +35,11 @@ int main(int argc, char *argv[])
     if (rf.check("help"))
     {
         cout<<"Options"<<endl;
-        cout<<"\t--robot                <robot-name>: robot name"<<endl;
```

Comment:
`yarprobotstatepublisher` is a relatively stable tool, that is used by several users of iDynTree, so we can't simply change the name of an option, but we should first deprecated it (by still supporting it in addition to the new option) but printing a warning message, and then remove it after one stable release (so this change will be released in 0.12, and we can remove it for the 0.13 release). See https://github.com/robotology/yarp/blob/master/.github/CONTRIBUTING.md#deprecated-features for some related docs.

Reply:
I don't see any standard message for deprecated options, so I am just printing the warning with `std::cerr` (https://github.com/robotology/idyntree/pull/538/commits/a29e145bf2859d7da46dd479339b5bbcd557e26f). 

@traversaro let me know if there is a prefered way to do that.

Reply:
Printing a simple warning with `std::cerr` is fine.

Reply:
> Printing a simple warning with `std::cerr` is fine.

ok, then its done in https://github.com/robotology/idyntree/commit/a29e145bf2859d7da46dd479339b5bbcd557e26f